identity/src/test/java/com/android/identity/MobileSecurityObjectGeneratorTest.java

```diff
@@ -0,0 +1,331 @@
+        Assert.assertEquals("B68C8AFCB2AAF7C581411D2877DEF155BE2EB121A42BC9BA5B7312377E068F66"
+                .toLowerCase(), Util.toHex(isoDigestIDs.get(8L)));
+        Assert.assertEquals("0B3587D1DD0C2A07A35BFB120D99A0ABFB5DF56865BB7FA15CC8B56A66DF6E0C"
+                .toLowerCase(), Util.toHex(isoDigestIDs.get(9L)));
+        Assert.assertEquals("C98A170CF36E11ABB724E98A75A5343DFA2B6ED3DF2ECFBB8EF2EE55DD41C881"
+                .toLowerCase(), Util.toHex(isoDigestIDs.get(10L)));
+        Assert.assertEquals("B57DD036782F7B14C6A30FAAAAE6CCD5054CE88BDFA51A016BA75EDA1EDEA948"
+                .toLowerCase(), Util.toHex(isoDigestIDs.get(11L)));
+        Assert.assertEquals("651F8736B18480FE252A03224EA087B5D10CA5485146C67C74AC4EC3112D4C3A"
+                .toLowerCase(), Util.toHex(isoDigestIDs.get(12L)));
+    }
+
+    private void checkISOUSDigest(Map<Long, byte[]> isoUSDigestIDs) {
+        Assert.assertEquals(Set.of(0L, 1L, 2L, 3L), isoUSDigestIDs.keySet());
+        Assert.assertEquals("D80B83D25173C484C5640610FF1A31C949C1D934BF4CF7F18D5223B15DD4F21C"
+                .toLowerCase(), Util.toHex(isoUSDigestIDs.get(0L)));
+        Assert.assertEquals("4D80E1E2E4FB246D97895427CE7000BB59BB24C8CD003ECF94BF35BBD2917E34"
+                .toLowerCase(), Util.toHex(isoUSDigestIDs.get(1L)));
+        Assert.assertEquals("8B331F3B685BCA372E85351A25C9484AB7AFCDF0D2233105511F778D98C2F544"
+                .toLowerCase(), Util.toHex(isoUSDigestIDs.get(2L)));
+        Assert.assertEquals("C343AF1BD1690715439161ABA73702C474ABF992B20C9FB55C36A336EBE01A87"
+                .toLowerCase(), Util.toHex(isoUSDigestIDs.get(3L)));
+    }
+
+    public void testFullMSO(String digestAlgorithm) {
+        PublicKey deviceKeyFromVector = Util.getPublicKeyFromIntegers(
+                new BigInteger(TestVectors.ISO_18013_5_ANNEX_D_STATIC_DEVICE_KEY_X, 16),
+                new BigInteger(TestVectors.ISO_18013_5_ANNEX_D_STATIC_DEVICE_KEY_Y, 16));
+        final Timestamp signedTimestamp = Timestamp.ofEpochMilli(1601559002000L);
+        final Timestamp validFromTimestamp = Timestamp.ofEpochMilli(1601559002000L);
+        final Timestamp validUntilTimestamp = Timestamp.ofEpochMilli(1633095002000L);
+        final Timestamp expectedTimestamp = Timestamp.ofEpochMilli(1611093002000L);
+
+        Map<String, List<String>> deviceKeyAuthorizedDataElements = new HashMap<>();
+        deviceKeyAuthorizedDataElements.put("a", List.of("1", "2", "f"));
+        deviceKeyAuthorizedDataElements.put("b", List.of("4", "5", "k"));
+
+        Map<Integer, byte[]> keyInfo = new HashMap<>();
+        keyInfo.put(10, Util.fromHex("C985"));
+
+        byte[] encodedMSO = new MobileSecurityObjectGenerator(digestAlgorithm,
+                "org.iso.18013.5.1.mDL", deviceKeyFromVector)
+                .addDigestIDs("org.iso.18013.5.1", generateISODigest())
+                .addDigestIDs("org.iso.18013.5.1.US", generateISOUSDigest())
```

Comment:
I think we also need to pass `digestAlgorithm` to these so the right size digests are inserted. Instead of hardcoding hashes in those functions (and the `check*()` ones too), maybe just do something like this

```
MessageDigest digester = MessageDigest.getInstance(digestAlgorithm);
byte[] hash = digester.digest("foo".getBytes(StandardCharsets.UTF_8));
```